tradebot.py

```diff
@@ -0,0 +1,345 @@
+      try:
+        args[3] = args[3].upper()
+        order = api.submit_order(args[3],args[2],args[1],args[0].lower(),args[4],limit_price=args[5],stop_price=args[6])
+        text = f'Stop-Limit order of | {args[1]} {args[2]} {args[3]} at stop price {args[6]} and limit price {args[5]} | submitted.  Order id = {order.id}.'
+        response = requests.post(url="https://slack.com/api/chat.postMessage",data={
+          "token": SLACK_TOKEN,
+          "channel": request.form.get("channel_name"),
+          "text": text
+        })
+      except Exception as e:
+        response = requests.post(url=request.form.get("response_url"), data=json.dumps({"text": f"ERROR: {str(e)}"}), headers={"Content-type": "application/json"})
+    else:
+      return BAD_ARGS
+  asyncio.run(sub_order(api,request,args))
+  return ""
+
+
+# Lists certain things.  Must contain 1 argument: orders, positions, or streams.
+@app.route("/list",methods=["POST"])
+def list_handler():
+  args = request.form.get("text").split(" ")
+  if(len(args) == 0):
+    return WRONG_NUM_ARGS
+  if(args[0] == "positions"):
+    try:
+      positions = api.list_positions()
+      if(len(positions) == 0):
+        return "No positions."
+      positions = map(lambda x: (f'Symbol: {x.symbol}, Qty: {x.qty}, Side: {x.side}, Entry price: {x.avg_entry_price}, Current price: {x.current_price}'),positions)
+      return "Listing positions...\n" + '\n'.join(positions)
+    except Exception as e:
+      response = requests.post(url=request.form.get("response_url"), data=json.dumps({"text": f"ERROR: {str(e)}"}), headers={"Content-type": "application/json"})
+  elif(args[0] == "orders"):
+    try:
+      orders = api.list_orders(status="open")
+      if(len(orders) == 0):
+        return "No orders."
+      orders = map(lambda x: (f'Symbol: {x.symbol}, Qty: {x.qty}, Side: {x.side}, Type: {x.type}, Amount filled: {x.filled_qty}{(f", Stop Price = {x.stop_price}","")[x.stop_price == None]}{(f", Limit Price = {x.limit_price}","")[x.limit_price == None]}'),orders)
+      return "Listing orders...\n" + '\n'.join(orders)
+    except Exception as e:
+      response = requests.post(url=request.form.get("response_url"), data=json.dumps({"text": f"ERROR: {str(e)}"}), headers={"Content-type": "application/json"})
+  elif(args[0] == "streams"):
+    text = "Listing active streams...\n"
+    try:
+      for stream in streams:
+        if(streams[stream] != None):
+          text += (stream + "\n")
+      if(text == "Listing active streams...\n"):
+        return "No active streams."
+      return text
+    except Exception as e:
+      response = requests.post(url=request.form.get("response_url"), data=json.dumps({"text": f"ERROR: {str(e)}"}), headers={"Content-type": "application/json"})
+  else:
+    return BAD_ARGS
+
+# Clears positions or orders.  Must contain 1 argument: positions or orders
```

Comment:
Just a suggestion, maybe add an "all" option? Then in the code below, say "if args[0] == 'orders' or args[0] == 'all': ... if args[0] == 'positions' or args[0] == 'all':". Seems like it.might be useful if someone wanted to shut down everything.

Reply:
Tried implementing it, decided to scrap it bc when clearing positions post-market it places orders instead, and the user might get confused on why there are still orders after clearing everything